components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -449,6 +449,23 @@ protected Map<ClaimMapping, String> getSubjectAttributes(OAuthClientResponse tok
     protected void initiateAuthenticationRequest(HttpServletRequest request, HttpServletResponse response,
                                                  AuthenticationContext context) throws AuthenticationFailedException {
 
+        try {
+            String loginPage = prepareLoginPage(request, context);
+            response.sendRedirect(loginPage);
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
```

Comment:
Why do we need build the diagnostic builder when refactoring?

Reply:
Addressed with https://github.com/wso2-extensions/identity-outbound-auth-oidc/pull/177/commits/784af7df50b7baf09eceb2341ddb4cae6f5d0475